external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala

```diff
@@ -136,6 +137,20 @@ class KafkaTestUtils(
     kdcConf.setProperty(MiniKdc.DEBUG, "true")
     kdc = new MiniKdc(kdcConf, kdcDir)
     kdc.start()
+    val krb5Conf = Source.fromFile(kdc.getKrb5conf, "UTF-8").getLines()
+    val rewriteKrb5Conf = krb5Conf.map(s => if (s.contains("libdefaults")) {
```

Comment:
If `libdefaults` doesn't exist then no need to add `default_tkt_enctypes`, etc..? (tests would show the intention)

Reply:
> If `libdefaults` doesn't exist then no need to add `default_tkt_enctypes`, etc..? (tests would show the intention)

Emmm in krb5.conf, `libdefault` is stable.

Reply:
It's stable but MiniKDC can create a `krb5.conf` file which doesn't contain `libdefault`. Such case the code does something different.

Reply:
> It's stable but MiniKDC can create a `krb5.conf` file which doesn't contain `libdefault`. Such case the code does something different.

emmmm, Stable in `MiniKdc`

Reply:
Just to get to the same page based on what do you say this? Please point to the doc of this statement.

My findings:
* This description is coming from the `krb5.conf` file format:
```
The krb5.conf file may contain the following sections:

[libdefaults] | Settings used by the Kerberos V5 library
...
```

* In `MiniKdc` until now haven't seen any guarantee that it will contain such section.

Reply:
![image](https://user-images.githubusercontent.com/46485123/69860455-4a2ab880-12d1-11ea-88df-9eeea69ac268.png)
![image](https://user-images.githubusercontent.com/46485123/69860463-4f880300-12d1-11ea-812d-dd54453d9837.png)

Reply:
@gaborgsomogyi 
It generate `krb5.conf` from a template file and replace related key.

Reply:
And we want to do this the same way

Reply:
OK, so make things more simple. I've double checked 2.7.4 code and the template is in the hadoop library. If the hadoop lib removes the `libdefaults` section because you haven't pointed any guarantee that it will remain, do this test will fail with the actual stand of the PR? If it fails the guy who will fix this must spend quite a time to understand what's going on...

Reply:
> OK, so make things more simple. I've double checked 2.7.4 code and the template is in the hadoop library. If the hadoop lib removes the `libdefaults` section because you haven't pointed any guarantee that it will remain, do this test will fail with the actual stand of the PR? If it fails the guy who will fix this must spend quite a time to understand what's going on...

Yea... so build spark's own MiniKDC will make us more clear about kerberos config...
Understanding  what we have shown in this pr ... really need to know kerberos's config a lot ..